README.md

```diff
@@ -4,35 +4,30 @@ This repo contains the new prototype for a new compute-centric 2D GPU renderer.
 
 It succeeds the previous prototype, [piet-metal].
 
+The latest version is a middleware for [`wgpu`]. This is used as the rendering backend for
+[xilem], a UI toolkit.
+
+<!-- TODO: Are we transitioning to more production? If so, should we rewrite the README a bit? -->
+
 ## Goals
 
 The main goal is to answer research questions about the future of 2D rendering:
 
-* Is a compute-centered approach better than rasterization ([Direct2D])? How much so?
+-   Is a compute-centered approach better than rasterization ([Direct2D])? How much so?
 
-* To what extent do "advanced" GPU features (subgroups, descriptor arrays) help?
+-   To what extent do "advanced" GPU features (subgroups, descriptor arrays) help?
 
-* Can we improve quality and extend the imaging model in useful ways?
-
-Another goal is to explore a standards-based, portable approach to GPU compute.
+-   Can we improve quality and extend the imaging model in useful ways?
 
 ## Blogs and other writing
 
 Much of the research progress on piet-gpu is documented in blog entries. See [doc/blogs.md](doc/blogs.md) for pointers to those.
 
 There is a much larger and detailed [vision](doc/vision.md) that explains the longer-term goals of the project, and how we might get there.
 
-### Why not gfx-hal?
-
-It makes a lot of sense to use gfx-hal, as it addresses the ability to write kernel and runtime code once and run it portably. But in exploring it I've found some points of friction, especially in using more "advanced" features. To serve the research goals, I'm enjoying using Vulkan directly, through [ash], which I've found does a good job tracking Vulkan releases. One example is experimenting with `VK_EXT_subgroup_size_control`.
-
-The hal layer in this repo is strongly inspired by gfx-hal, but with some differences. One is that we're shooting for a compile-time pipeline to generate GPU IR on DX12 and Metal, while gfx-hal ships [SPIRV-Cross] in the runtime. To access [Shader Model 6], that would also require bundling [DXC] at runtime, which is not yet implemented (though it's certainly possible).
-
-### Why not wgpu?
-
-The case for wgpu is also strong, but it's even less mature. I'd love to see it become a solid foundation, at which point I'd use it as the main integration with [Druid].
+## History
 
-In short, the goal is to facilitate the research now, collect the data, and then use that to choose a best path for shipping later.
+A prior incarnation used a custom cross-API hal. An archive of this version can be found in the branches [`custom-hal-archive-with-shaders`] and [`custom-hal-archive`].
```

Comment:
I'm not sure we should keep branches around, it clutters things up. What I've taken to doing is creating a PR and then closing it. The branch is still accessible through the PR. That said, it does make us more dependent on GitHub specifically rather than Git.

Reply:
I've put some thoughts in Zulip; I think we can resolve this outside of this PR.